pkg/cloudprovider/aws/constraints.go

```diff
@@ -39,23 +43,67 @@ var (
 	SubnetTagKeyLabel        = AWSLabelPrefix + "subnet-tag-key"
 	SecurityGroupNameLabel   = AWSLabelPrefix + "security-group-name"
 	SecurityGroupTagKeyLabel = AWSLabelPrefix + "security-group-tag-key"
-	AllowedLabels            = []string{
-		CapacityTypeLabel,
-		LaunchTemplateNameLabel,
-		SubnetNameLabel,
-		SubnetTagKeyLabel,
-		SecurityGroupNameLabel,
-		SecurityGroupTagKeyLabel,
-	}
-	AWSToKubeArchitectures = map[string]string{
+	AWSToKubeArchitectures   = map[string]string{
 		"x86_64":                   v1alpha3.ArchitectureAmd64,
 		v1alpha3.ArchitectureArm64: v1alpha3.ArchitectureArm64,
 	}
 	KubeToAWSArchitectures = functional.InvertStringMap(AWSToKubeArchitectures)
 )
 
-// Constraints are AWS specific constraints
-type Constraints v1alpha3.Constraints
+func NewConstraints(constraints *v1alpha3.Constraints) *Constraints {
+	extensions := &Extensions{}
+	if err := constraints.UnmarshalExtensions(extensions); err != nil {
+		panic(fmt.Errorf("Failed to parse constraints, %w", err))
+	}
+	return &Constraints{
+		Constraints: constraints,
+		Extensions:  extensions,
+	}
+}
+
+// Constraints are used to specify node creation parameters.
+type Constraints struct {
+	*v1alpha3.Constraints
+	*Extensions
+}
+
+// Extensions are parameters specific to this cloud provider
+type Extensions struct {
+	// Cluster is used to connect Nodes to the Kubernetes cluster.
+	// +required
+	Cluster Cluster `json:"cluster"`
+	// InstanceRole is the AWS identity that instances use.
+	// +required
+	InstanceRole string `json:"instanceRole"`
+	// Subnets specify the names of the subnets.
+	// +optional
+	Subnets []string `json:"subnets,omitempty"`
+	// SubnetSelector discovers subnets by tags. A value of "" is a wildcard.
+	// +optional
+	SubnetSelector map[string]string `json:"subnetTags,omitempty"`
+	// SecurityGroups specify the names of the security groups.
+	// +optional
+	SecurityGroups []string `json:"securityGroups,omitempty"`
+	// SecurityGroupSelector discovers security groups by tags. A value of "" is a wildcard.
+	// +optional
+	SecurityGroupsSelector map[string]string `json:"securityGroupSelector,omitempty"`
+	// LaunchTemplate for the node. If not specified, a launch template will be generated.
+	// +optional
+	LaunchTemplate *string `json:"launchTemplate,omitempty"`
+	// CapacityType for the node. If not specified, defaults to on-demand
+	// +optional
+	CapacityType *string `json:"capacityType,omitempty"`
+}
+
+// Cluster configures the cluster that the provisioner operates against.
+type Cluster struct {
```

Comment:
Why is there not longer a `CABundle` field?

Reply:
For AWS, we've decided to always discover the CABundle using the rest.Config.